test/regressions/TestViewer.js

```diff
@@ -77,7 +84,7 @@ function TestViewer(props) {
             <JoyBox
               aria-busy={!ready}
               data-testid="testcase"
-              sx={{ bgcolor: 'background.body', display: 'inline-block', p: 1 }}
+              sx={{ bgcolor: 'background.body', ...viewerBoxSx }}
```

Comment:
would this be enough?

```suggestion
              sx={{ bgcolor: 'background.body', display: 'block', p: 1 }}
```

Reply:
> would this be enough?

Yes, it has the same effect. I'm indifferent to the exact solution, as long as it's not `inline` 🙂

Reply:
Are we ok with docs demos potentially looking slightly different to their corresponding regression screenshots? If that's ok `block` is fine. Probably the demo wrapper should wrap the contents in a `<div>` so the demo contents are not affected by `flex`, although `flex` can be a good thing to help build more robust components e.g. https://app.argos-ci.com/mui/material-ui/builds/31842/107814095 breaks because it has several top-level elements instead of just one.

Reply:
> Are we ok with docs demos potentially looking slightly different to their corresponding regression screenshots?

There's a nuance here that not all screenshots are originating from docs demos. Some come from the templates. And I believe there's a set of separately curated ones as well. This is what made me open this PR in the first place as the templates behave flaky when rendered in an inline container. I think in light of that, `block` makes sense. But I'm personally ok either way, but let's not waste too much time on this, the idea is to get away from `inline-block`, the difference between `flex` and `block` is far from as impactful as that.